src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts

```diff
@@ -2,23 +2,53 @@ import { Pipe, PipeTransform } from '@angular/core';
 
 @Pipe({ name: 'artemisDurationFromSeconds' })
 export class ArtemisDurationFromSecondsPipe implements PipeTransform {
+    private readonly secondsInDay = 60 * 60 * 24;
+    private readonly secondsInHour = 60 * 60;
+    private readonly secondsInMinute = 60;
+
     /**
-     * Convert seconds to a human-readable duration format (mm:ss).
+     * Convert seconds to a human-readable duration format:
+     * If short is false: "1d 11h 7min 6s", where the days and hours are left out if their value and all higher values are zero
+     * If short is true: "xx unit yy unit", where only the two highest units are shown. If the time is between 10 minutes and one hour, only the minutes are shown
+     * @param short? {boolean} allows the format to be shortened
      * @param seconds {number}
      */
-    transform(seconds: number): string {
-        const minutes = Math.floor(seconds / 60);
-        seconds = seconds - minutes * 60;
+    transform(seconds: number, short?: boolean): string {
+        if (seconds === null || seconds <= 0) {
+            return '0min 0s';
+        }
 
-        let minutesOut = minutes.toString();
-        if (minutes < 10) {
-            minutesOut = '0' + minutes.toString();
+        const days = Math.floor(seconds / this.secondsInDay);
+        const hours = Math.floor((seconds % this.secondsInDay) / this.secondsInHour);
+        const minutes = Math.floor((seconds % this.secondsInHour) / this.secondsInMinute);
```

Comment:
I remember something such as "Please use [`Math.round`](https://github.com/ls1intum/Artemis/blob/fe5ab6c42bb64c9d6eabef8b58a131b37da3db80/src/main/webapp/app/shared/util/utils.ts#L60) from our `utils.ts` file" - I think this might apply here as well?

Reply:
As long as I just want to round to the next integer, Math.round is having exactly the way as round() in utils.ts behaves.
(See first lines in that method)